xarray_subset_grid/grids/ugrid.py

```diff
@@ -154,14 +154,15 @@ def subset_polygon(
         # NOTE: UGRIDS can be zero-indexed OR one-indexed!
         #       see the UGRID spec.
         tris = face_node_connectivity - face_node_start_index
-        tri_mask = node_inside[tris]
+        valid_tris = tris.where(tris >= 0, drop=False).astype(int)
```

Comment:
It might be better to specify the precision: int32 should do it  (`np.int32`)?

And what will this do with NaN?

I still think it's better to pre-process, but this is a small lift, so ...

Reply:
never mind -- nos I see: `face_node_connectivity.fillna(-1)` --but still the int type issue

Reply:
See the next commit, the nan is handled by `fillna`

Reply:
resolved in latest commit!